### PR TITLE
Issue 1987: Updated pravega.io SegmentStore documentation

### LINKED ISSUE #1987: Segment store documentation
**Problem description**
We have a doc for the segment store under `docs/segment-store-service.md`. I don't see it linked anywhere in the pravega.io doc section. We should link and update the document.

**Problem location**
Documentation.

**Suggestions for an improvement**
Update documentation for the segment store.

### PR DESCRIPTION
**Pull request by andreipaduroiu**
**Change log description**
Updated Segment Store documentation. Linked it from Pravega Concepts.

**Purpose of the change**
Fixes #1987.

**How to verify it**
Read documentation.

### CODE REVIEW

File: docs/segment-store-service.md [hunk 1]
```diff
@@ -30,14 +30,15 @@ Throughout the rest of this document, we will use the following terminology:
 
 # Architecture
 The **Segment Store** is made up of the following components:
-- **Pravega Node**: a host running a Pravega Process.
-- **Stream Segment Container** (or **Segment Container**): A logical grouping of Stream Segments. The mapping of Segments to Containers is deterministic and does not require any persistent store; Segments are mapped to Containers via a hash function (based on the Segment's name).
-- **Durable Data Log Adapter**: an abstraction layer for Tier-1 Storage.
-- **Storage Adapter**: an abstraction layer for Tier-2 Storage.
-- **Cache**: an abstraction layer for append data caching.
-- **Streaming Client**: an API that can be used to communicate with the Pravega Segment Store. 
+- `Pravega Node`: a host running a Pravega Process.
```
Comment (fpj) [inline docs/segment-store-service.md]: Let's clear up the convention we are using for formatting the text. Here is my suggestion:

- For anything that comes from code, e.g., a method `isComplete` should have back-ticks around it.
- For any term that we define, we italicize when we define, e.g., the _segment store_ implements the control plane of Pravega and consists of one or more servers. Further occurrences of the term do no need to be italicized.
- For any term that we would like to highlight in the text, but that is not defined by us, then we make it *bold*.

File: docs/segment-store-service.md [hunk 2]
```diff
@@ -30,14 +30,15 @@ Throughout the rest of this document, we will use the following terminology:
 
 # Architecture
```
Comment (fpj) [inline docs/segment-store-service.md]: You had some nice illustrations that we could add here. Please consider adding and discussing them.

Comment (andreipaduroiu) [reply docs/segment-store-service.md]: The illustrations are still there.

File: docs/segment-store-service.md (hunk 1 above)
Comment (andreipaduroiu) [reply docs/segment-store-service.md]: Thanks for the suggestions. I updated the doc.

**Pull request merged**